extensions/sql-database-projects/src/controllers/databaseProjectTreeViewProvider.ts

```diff
@@ -71,7 +78,13 @@ export class SqlDatabaseProjectTreeViewProvider implements vscode.TreeDataProvid
 		const projNode = this.roots.find(x => x instanceof ProjectRootTreeItem ? (<ProjectRootTreeItem>x).project === project : false);
 
 		if (projNode) {
-			this.treeView?.reveal(projNode, { focus: true, expand: true });
+			await this.treeView?.reveal(projNode, { focus: true, expand: true });
+		}
+	}
+
+	public async expandAllNodes(): Promise<void> {
```

Comment:
is there a way to make this so that only the selected project gets expanded instead of all the open projects?

Reply:
yes, is that the desired behavior of "Expand All" / should the action be renamed in that case?

Reply:
hmm.. good point. I think it would be okay to track per project expand collapse as seperate item though - right?

Reply:
yup, would be fine to add them separately.  Just gotta keep that in mind when designing this portion so adding the next step is relatively easy!

Reply:
New issue opened to track per project expand/collapse: #11795